engine/crates/federated-graph/src/render_sdl/display_utils.rs

```diff
@@ -1,12 +1,28 @@
 use crate::*;
-use std::{
-    fmt::{self, Display, Write},
-    iter,
-};
+use std::fmt::{self, Display, Write};
 
 pub(super) const BUILTIN_SCALARS: &[&str] = &["ID", "String", "Int", "Float", "Boolean"];
 pub(super) const INDENT: &str = "    ";
 
+/// Lets you take a routine that expects a formatter, and use it on a string.
+pub(in crate::render_sdl) fn with_formatter<F>(out: &mut String, action: F) -> fmt::Result
```

Comment:
This is fairly basic, there should be something standard to do that, am I forgetting something?

Reply:
aren't you re-implementing https://doc.rust-lang.org/std/macro.write.html ?

Reply:
It's basically a version of `write!()` that instead of `Display` arguments, takes a function that takes a `Formatter`.

Reply:
>that instead of Display arguments, takes a function that takes a Formatter.

Isn't that the definition of `impl std::fmt::Display`?

Reply:
ok got it

Reply:
Basically I could use `write()` with small structs that implement `Display`, but I'd rather write a function than a struct + a display impl if it's going to happen 20 times in the same module.

Reply:
That function will become unnecessary once render_federated_sdl() works like render_api_sdl(), with just one Display impl at the top.